### PR TITLE
feat(python): Migrate llama-index example to v0.10+ 

### LINKED ISSUE #206: [Python] Migrate llama-index example to llama-index 0.10
- [ ] run `npx create-llama`
- [ ] add instrumentation.py and add all necessary depenancies
- [ ] setup docker compose with phoenix
- [ ] Update README and tutorials

### PR DESCRIPTION
**Pull request by pbadhe**
Resolves #206

### CODE REVIEW

**Comment by pbadhe** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by axiomofjoy** [comment]
@pbadhe Let me know when this is ready for review!

**Comment by pbadhe** [comment]
@axiomofjoy It's ready...

**Comment by axiomofjoy** [comment]
Thanks @pbadhe, will take a look.

File: python/examples/llama-index/backend/pyproject.toml [hunk 1]
```diff
@@ -1,22 +1,24 @@
+[tool]
```
Comment (axiomofjoy) [inline python/examples/llama-index/backend/pyproject.toml:1]: Is this line necessary?

File: python/examples/llama-index/backend/app/engine/loaders/__init__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+def get_documents():
+    documents = []
+    config = load_configs()
+    for loader_type, loader_config in config.items():
+        logger.info(f"Loading documents from loader: {loader_type}, config: {loader_config}")
+        match loader_type:
+            case "file":
+                document = get_file_documents(FileLoaderConfig(**loader_config))
+            case _:
+                raise ValueError(f"Invalid loader type: {loader_type}")
```
Comment (axiomofjoy) [inline python/examples/llama-index/backend/app/engine/loaders/__init__.py:25]: This Python syntax was introduced in Python 3.10. If we haven't already, let's put a warning in the README that this project requires Python 3.10 or later to run from source (not to run with Docker compose, though).

File: python/examples/llama-index/backend/app/settings.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+from llama_index.core.settings import Settings
+from llama_index.embeddings.openai import OpenAIEmbedding
+from llama_index.llms.openai import OpenAI
+
+
+def llm_config_from_env() -> Dict:
+    from llama_index.core.constants import DEFAULT_TEMPERATURE
+
+    model = os.getenv("MODEL", "gpt-3.5-turbo-0125")
```
Comment (axiomofjoy) [inline python/examples/llama-index/backend/app/settings.py:12]: Shall we bump the default up to "gpt-4" or later?

**Review by axiomofjoy — COMMENTED** [review]
@pbadhe I'm noticing a few additions to the frontend that don't seem to be used anywhere (e.g., for code blocks, file uploads, etc.). Are these changes coming from `create-llama`?

**Comment by axiomofjoy** [comment]
Looks good overall @pbadhe. Just had a few non-blocking suggestions and a question around some of the unused components in the frontend directory. If those changes are coming from `create-llama`, no worries, let's just merge them in.

**Comment by axiomofjoy** [comment]
Tested this out, works for me.

File: python/examples/llama-index/backend/pyproject.toml (hunk 1 above)
Comment (pbadhe) [reply python/examples/llama-index/backend/pyproject.toml:1]: It's not, will remove it.

File: python/examples/llama-index/backend/app/engine/loaders/__init__.py (hunk 2 above)
Comment (pbadhe) [reply python/examples/llama-index/backend/app/engine/loaders/__init__.py:25]: Yes, you're right, I had to setup an env with py ^3.11 as specified in .toml file.

File: python/examples/llama-index/backend/app/settings.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    config = {
+        "model": model,
+        "temperature": float(temperature),
+        "max_tokens": int(max_tokens) if max_tokens is not None else None,
+    }
+    return config
+
+
+def embedding_config_from_env() -> Dict:
+    model = os.getenv("EMBEDDING_MODEL", "text-embedding-ada-002")
```
[the remark this answers is not in the document]
Comment (pbadhe) [reply python/examples/llama-index/backend/app/settings.py:25]: Yep, will change it to `text-embedding-3-small`, less price more performance.

File: python/examples/llama-index/backend/app/settings.py (hunk 3 above)
Comment (pbadhe) [reply python/examples/llama-index/backend/app/settings.py:12]: I won't be the best person to answer. My opinion, gpt-4.

Comment (axiomofjoy) [reply python/examples/llama-index/backend/app/settings.py:12]: Let's roll with `gpt-4-turbo-preview`.

**Comment by pbadhe** [comment]
> @pbadhe I'm noticing a few additions to the frontend that don't seem to be used anywhere (e.g., for code blocks, file uploads, etc.). Are these changes coming from `create-llama`?

Yes @axiomofjoy, they are all coming from create-llama. I assume the file upload options are specifically for the case when you are using just Next.js, but here we have a FastAPI python backend too. I did try to remove the unwanted backend code as much as I could, for ex: non-streaming endpoints, embedding from db, etc.

**Comment by axiomofjoy** [comment]
> > @pbadhe I'm noticing a few additions to the frontend that don't seem to be used anywhere (e.g., for code blocks, file uploads, etc.). Are these changes coming from `create-llama`?
> 
> 
> 
> Yes @axiomofjoy, they are all coming from create-llama. I assume the file upload options are specifically for the case when you are using just Next.js, but here we have a FastAPI python backend too. I did try to remove the unwanted backend code as much as I could, for ex: non-streaming endpoints, embedding from db, etc.

Thanks @pbadhe. Don't worry about removing code. Looks good.

**Review by axiomofjoy — APPROVED** [review]

**Comment by axiomofjoy** [comment]
@pbadhe Let me know when you're ready to merge.

**Comment by pbadhe** [comment]
@axiomofjoy Thanks for the quick reviews! Ready to merge.

**Comment by axiomofjoy** [comment]
CI failures are independent of this PR. Merging.

**Comment by axiomofjoy** [comment]
Thanks @pbadhe!

**Pull request merged**